File: app/src/main/java/com/chuongvd/app/signal/data/source/remote/datasource/IUserRemoteDataSource.java
package com.chuongvd.app.signal.data.source.remote.datasource;

import com.chuongvd.app.signal.data.source.remote.request.LoginRequest;
import com.chuongvd.app.signal.data.entity.UserEntity;
import io.reactivex.Observable;

public interface IUserRemoteDataSource {
    Observable<UserEntity> login(LoginRequest request);
}

File: app/src/main/java/com/chuongvd/app/signal/data/source/remote/datasource/IUserRemoteDataSource.kt
package com.chuongvd.app.signal.data.source.remote.datasource

import com.chuongvd.app.signal.data.entity.UserEntity
import com.chuongvd.app.signal.data.source.remote.request.LoginRequest
import io.reactivex.Observable

interface IUserRemoteDataSource {
  fun login(request: LoginRequest): Observable<UserEntity>
}
